user-operator/src/main/java/io/strimzi/operator/user/Main.java

```diff
@@ -70,7 +73,9 @@ public static void main(String[] args) {
         Security.setProperty("networkaddress.cache.ttl", dnsCacheTtl);
 
         // Create and log UserOperatorConfig
-        UserOperatorConfig config = UserOperatorConfig.fromMap(System.getenv());
+        Map<String, String> m = new HashMap<>(System.getenv());
+        m.keySet().retainAll(UserOperatorConfig.keyNames());
```

Comment:
If this is needed, why does it not live in the constructor or in the `buildFromMap` method?